package/MDAnalysis/analysis/hbond_analysis.py

```diff
@@ -0,0 +1,199 @@
+class from the module mod:`MDAnalysis.analysis.hbonds.hbond_analysis`, which itself was modeled after the `VMD
+HBONDS plugin`_.
+
+.. _`VMD HBONDS plugin`: http://www.ks.uiuc.edu/Research/vmd/plugins/hbonds/
+
+
+Options:
+  - *d_h_cutoff* (Å) [1.2] : Distance cutoff used for finding donor-hydrogen pairs
+  - *d_a_cutoff* (Å) [3.0] : Distance cutoff for hydrogen bonds. This cutoff refers to the D-A distance.
+  - *d_h_a_angle_cutoff* (degrees) [150] : D-H-A angle cutoff for hydrogen bonds.
+
+
+Output:
+  - *frame* : frame at which a hydrogen bond was found
+  - *donor id* : atom id of the hydrogen bond donor atom
+  - *hydrogen id* : atom id of the hydrogen bond hydrogen atom
+  - *acceptor id* : atom id of the hydrogen bond acceptor atom
+  - *distance* (Å): length of the hydrogen bond
+  - *angle* (degrees): angle of the hydrogen bond
+
+Hydrogen bond data are returned in a :class:`numpy.ndarray` on a "one line, one observation" basis
+and can be accessed via :attr:`HydrogenBondAnalysis.hbonds`::
+
+    results = [
+        [
+            <frame>,
+            <donor index (0-based)>,
+            <hydrogen index (0-based)>,
+            <acceptor index (0-based)>,
+            <distance>,
+            <angle>
+        ],
+        ...
+    ]
+
+Classes
+-------
+
+.. autoclass:: HydrogenBondAnalysis
+   :members:
+
+   .. attribute:: hbonds
+
+"""
+
+import numpy as  np
+
+from . import base
+from MDAnalysis.lib.distances import capped_distance
+from MDAnalysis.lib.distances import calc_angles
+
+class HydrogenBondAnalysis(base.AnalysisBase):
+    """
+    Perform an analysis of hydrogen bonds in a Universe.
+    """
+
+    def __init__(self, universe, donors_sel, hydrogens_sel, acceptors_sel, d_h_cutoff, d_a_cutoff, d_h_a_angle_cutoff):
+        """
+        Set up atom selections and geometric criteria for finding hydrogen bonds in a Universe.
+
+        :param universe : Universe
+            MDAnalysis Universe object
+        :param donors_sel : str
+            Selection string for the hydrogen bond donor atoms
+        :param hydrogens_sel :  str
+            Selection string for the hydrogen bond hydrogen atoms
+        :param acceptors_sel : str
+            Selection string for the hydrogen bond acceptor atoms
+        :param d_h_cutoff : float (optional)
+            Distance cutoff used for finding donor-hydrogen pairs [1.2]
+        :param d_a_cutoff : float (optional)
+            Distance cutoff for hydrogen bonds. This cutoff refers to the D-A distance. [3.0]
+        :param d_h_a_angle_cutoff: float (optional)
+            D-H-A angle cutoff for hydrogen bonds, in degrees. [150]
+
+        """
+
+        self.u = universe
+        self.donors_sel = donors_sel
+        self.hydrogens_sel = hydrogens_sel
+        self.acceptors_sel = acceptors_sel
+        self.d_h_cutoff = d_h_cutoff
+        self.d_a_cutoff = d_a_cutoff
+        self.d_h_a_angle = d_h_a_angle_cutoff
+
+
+    def run(self, start=None, stop=None, step=None):
```

Comment:
Instead of putting all logic into `run()` you should put create the `_single_frame()` method that works on a single frame. This will make it easier to, for instance, parallelize the code.

Reply:
I've fixed this.